var/spack/repos/builtin/packages/py-pythran/package.py

```diff
@@ -32,17 +33,21 @@ class PyPythran(PythonPackage):
 
     depends_on("py-setuptools", type="build")
     depends_on("py-ply@3.4:", type=("build", "run"))
-    depends_on("py-gast@0.5", when="@0.9.12:", type=("build", "run"))
-    depends_on("py-gast@0.4", when="@0.9.7:0.9.11", type=("build", "run"))
-    depends_on("py-gast@0.3.3:0.3", when="@0.9.6", type=("build", "run"))
-    depends_on("py-gast@0.3:", when="@0.9.4:0.9.5", type=("build", "run"))
-    depends_on("py-gast", when="@:0.9.3", type=("build", "run"))
+    depends_on("py-gast", type=("build", "run"))
+    # Unpin overly strict dependency requirements to ease concretization of TensorFlow
```

Comment:
@rgommers curious if you have any thoughts about this since these are adjacent to the scipy ecosystem. The latest version of TF requires gast 0.4, but only ancient versions of scipy/pythran/beniget support this, and those versions either don't build on some systems or can't be concretized for other reasons. It's quite frustrating that scipy/pythran/beniget don't support a range of dependency versions like every other package.

Reply:
P.S. Every version combination of pythran, beniget, and gast I tried built fine, so there should be absolutely no reason to pin any of these dependencies.

Reply:
Jk, things build fine, but crash at run time. Going to take a while to figure out exactly which versions are cross compatible...

Reply:
My understanding is that pythran is a build/link dependency, not a build dependency. I know that dependency type doesn't exist in pip, but in Spack we generally try to ensure that there is only one copy of each link/run dependency. @becker33 and @alalazo are working on loosening this restriction for build-only deps, but I don't think we can loosen this for link deps.

Reply:
I don't think that's the case. At least, it doesn't have to be. For SciPy, we only use Pythran as a `.py` -> `.cpp` transpiler and then use the generated sources as regular C++ ones. So there should not be a link dependency.

Reply:
You're right. Was that always the case? I feel like before the meson move, we linked to some pythran library or header.

Reply:
I think so - but it's quite hard to be sure. The `setup.py` build used `PythranExtension` to build extension modules. That is distutils-like and I'd be a little surprised if that inserted a shared library that is needed at runtime. But it's not impossible, so if you added it based on failures in CI that then disappeared, I guess there was such a thing.

Reply:
I can't test older versions of scipy on macOS due to older scipy not supporting newer pythran and older pythran not supporting newer macOS, so the mystery will always remain. But I'll submit a PR to make it a build dep, not build/link.